lms/db/_util.py

```diff
@@ -0,0 +1,24 @@
+from sqlalchemy import Select
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.orm import Query
+
+
+def compile_query(query: Query | Select, literal_binds: bool = True) -> str:
+    """
+    Return the SQL representation of `query` for postgres.
+
+    :param literal_binds: Whether or not replace the query parameters by their values.
+    """
+    if isinstance(query, Query):
+        # Support for SQLAlchemy 1.X style queryies, eg: db.query(Model).filter_by()
+        statement = query.statement
+    else:
+        # SQLALchemy 2.X style, eg: select(Model).where()
+        statement = query
+
+    return str(
+        statement.compile(
```

Comment:
That could be pretty useful.
Myself also used `mogrify` in the past which could handle more corner cases but at a cost of requiring access to a particular db cursor.